library/general/src/lib/installation/installation_data.rb

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "installation/installation_info"
+require "y2packager/medium_type"
+require "y2packager/product_reader"
+require "y2packager/resolvable"
+
+Yast.import "Mode"
+Yast.import "Pkg"
+
+module Installation
+  #
+  # Class for dumping the general installation data.
+  class InstallationData
+    # Register the callback for dumping the general installation data
+    def self.add
```

Comment:
np: To be honest, I do not like those classes with only "class methods". I would rather prefer a proper class, although that implies calling to `new`. But it might be just a matter of taste.